### PR TITLE
Weather coordinates

### PR DESCRIPTION
**Pull request by SallyYang02**
Added in the code (line 14-25) for all of the coordinates in the ASC2021 by making a nested list of all coordinates. Was wondering if there is a better way of calling each pair of coordinates (i.e. for loops) rather than manually indexing each and every point's lat and lon. Will add more code soon.

### CODE REVIEW

**Review by mdshiozaki — APPROVED** [review]
overall looks good with TODOs in place for the work that's coming. Looks fine to me

File: weather_get.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            i = int(i)
+            lat = row[i][0].strip()
+            lon = row[i][1].strip()
+            location.append(lat)
+            location.append(lon)
+            n = 2
+            locations = [location[j * n:(j + 1) * n] for j in range((len(location) + n - 1) // n )]  
+        #for point #1
+        lat = locations[0][0].strip()
+        lon = locations[0][1].strip()
```
Comment (e-wai) [inline weather_get.py:25]: I'm a tad bit confused around what's happening here? Assuming that from the CSV, we want a list like:

`[[lat, lon], [lat, lon], [lat, lon]]`

on which we're going to make API calls for each point?

Comment (e-wai) [reply weather_get.py:25]: In my head, the easiest way to do that is to declare an empty list before line 11, and for each row, use `.append([lat, lon])`

**Review by e-wai — CHANGES_REQUESTED** [review]

**Comment by e-wai** [comment]
Also, two quick things: did you mean to commit into the `weather-pairing` branch? Can we change the title of this PR to `[STRAT-#] Description`

File: weather_get.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+from config import WEATHER_API_KEY
+one_call_base = 'https://api.openweathermap.org/data/2.5/onecall?'
+path = os.path.join(os.path.dirname(__file__), '..', 'routes\ASC2021\ASC2021_draft.csv')
+with open(path, 'r') as wea2021:
+    for line in wea2021:
+        # read a set of latitude and longitude points
+        row = line.split(',')
+        location = [ ]
+        for i in range (len(row)):
+            i = int(i)
```
[the remark this answers is not in the document]
Comment (SallyYang02) [reply weather_get.py:16]: My IDE was giving me an error for not specifying i as an integer so I just it there as a precautionary measure. We can definitely remove that line if it works on everyone else's end.

File: weather_get.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+from config import WEATHER_API_KEY
+one_call_base = 'https://api.openweathermap.org/data/2.5/onecall?'
+path = os.path.join(os.path.dirname(__file__), '..', 'routes/ASC2021/ASC2021_draft.csv')
+location = [ ]
+with open(path, 'r') as wea2021:
+    for line in wea2021:
+        # read a set of latitude and longitude points
+        row = line.split(',')
+        for i in range (len(row)):
```
Comment (e-wai) [inline weather_get.py:16]: You can remove lines 16-18, and then move line 19 to below 23

File: weather_get.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+            lat = row[i][0].strip()
+            lon = row[i][1].strip()
+            location.append([lat,lon])
+        print(location)
```
Comment (e-wai) [inline weather_get.py:20]: I would move line 20 below whenever `append` is called

File: weather_get.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+path = os.path.join(os.path.dirname(__file__), '..', 'routes/ASC2021/ASC2021_draft.csv')
+location = [ ]
+with open(path, 'r') as wea2021:
+    for line in wea2021:
+        # read a set of latitude and longitude points
+        row = line.split(',')
+        for i in range (len(row)):
+            lat = row[i][0].strip()
+            lon = row[i][1].strip()
+            location.append([lat,lon])
```
Comment (e-wai) [inline weather_get.py:19]: `location.append([float(lat), float(lon)])`

File: weather_get.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        print(location)  
+        #for point #1
+        lat = locations[0][0].strip()
+        lon = row[0][1].strip()
```
Comment (e-wai) [inline weather_get.py:23]:
```suggestion
lat = row[0]
lon = row[1].strip()
```

**Review by e-wai — CHANGES_REQUESTED** [review]

**Review by e-wai — APPROVED** [review]
Looks good to me! 

I think eventually we should talk about how this will interact with other files (ie. we may need to wrap this in a function that calls it, saving into CSV is a related ticket as well) but we can keep that for future tickets 🎆

**Comment by e-wai** [comment]
Something weird happened here - I can't find this file in the repo, and it's not listed as having a file path. @ufshk @mdshiozaki am I going crazy?

**Comment by ufshk** [comment]
> 
> 
> Something weird happened here - I can't find this file in the repo, and it's not listed as having a file path. @ufshk @mdshiozaki am I going crazy?

It wasn't merged into master. That's why you can't find it in the repo.

It also does not have a filepath, it was put into the top-level directory on the branch it was merged into.

**Pull request merged**